api/search.go

```diff
@@ -67,6 +75,11 @@ func (api *SearchAPI) getSearch(w http.ResponseWriter, r *http.Request) {
 	// Get instanceID from datasetAPI
 	versionDoc, err := client.GetVersion(r.Context(), datasetID, edition, version)
 	if err != nil {
+		log.Error(err, nil)
+		if err := api.auditor.Record(r.Context(), "getSearch", "unsuccessful", auditParams); err != nil {
```

Comment:
Similarly to what I said to Nathan here: https://github.com/ONSdigital/dp-dataset-api

It would be better if you handled this with a goroutine and channels to reduce code duplication. Keep the "attempted" audit action as it is, but for the rest do something like this:

```go
auditEvent := make(chan string, 1)

go func() {
	action := <-auditEvent
	if auditErr := api.auditor.Record(r.Context(), getSearch, action, auditParams); auditErr != nil {
		handleAuditingFailure(w, auditErr, logData)
		return
	}
}()
```

```go
if err != nil {
	log.Error(err, logData)
	auditEvent <- actionUnsuccessful 
	handleObservationsErrorType(w, err)
	return
}
```

Define the goroutine immediately after you have made the `attempted` audit

Reply:
@mattrout92 cheers for the review. Have added added audit constants. 

Have left out the go-routine, good one to know (going in snippets) but causing issues with the unit tests (the expected calls to the auditor sporadically fail, race conditions I think). Search api is tiny enough there's minimal repetition so keeping it simple.